opensquirrel/passes/exporter/quantify_scheduler_exporter.py

```diff
@@ -85,26 +83,22 @@ def visit_controlled_gate(self, g: ControlledGate) -> None:
         raise UnsupportedGateError(g)
 
     def visit_measure(self, g: Measure) -> None:
-        qubit_index = g.qubit.index
-        bit_index = g.bit.index
-        acq_index = self.acq_index_record[qubit_index]
-        self.bit_string_mapping[bit_index] = (acq_index, qubit_index)
+        self.bit_string_mapping.add_measure(g)
+        acq_channel, acq_index = self.bit_string_mapping.get_last_added_acq()
         self.schedule.add(
             quantify_scheduler_gates.Measure(
                 self._get_qubit_string(g.qubit),
-                acq_channel=qubit_index,
+                acq_channel=acq_channel,
                 acq_index=acq_index,
                 acq_protocol="ThresholdedAcquisition",
             )
         )
-        self.acq_index_record[qubit_index] += 1
-        return
 
     def visit_reset(self, g: Reset) -> Any:
         self.schedule.add(quantify_scheduler_gates.Reset(self._get_qubit_string(g.qubit)))
 
 
-def export(circuit: Circuit) -> tuple[quantify_scheduler.Schedule, list[tuple[Any, Any]]]:
+def export(circuit: Circuit) -> tuple[quantify_scheduler.Schedule, dict[str, dict[str, int]]]:
```

Comment:
We should definitely use a type alias for `dict[str, dict[str, int]]]` (or more).

Reply:
Not in my opinion, no. Nothing stops us from giving a proper name to a `dict[str, dict[str, int]]`.